frontendGraph/src/main/java/com/virtuslab/gitmachete/frontend/graph/repository/RepositoryGraph.java

```diff
@@ -101,8 +101,16 @@ public int getNodeId(int nodeIndex) {
     assert nodeIndex < nodesCount() : "Bad nodeIndex: " + nodeIndex;
     return positionsOfVisibleEdges.get(nodeIndex).map(pos -> {
 
-      int downNodeIndex = nodeIndex + 1;
       int upNodeIndex = nodeIndex - 1;
+      int downNodeIndex = nodeIndex + 1;
+
+      // We can assume the following since we know that the first node (a root branch)
+      // AND the last node (some branch, possible indented child) has no visible edges in their rows.
+      // (The first condition is obvious. The second can be easily proved by contradiction.
+      // Suppose that the last node, at index n has a visible edge. Any visible edge has some (branch) node
+      // that it leads to, hence there must exists some node at index k > n being a target to the visible edge.
+      // But n is the index of the last node. Contradiction. )
+      assert upNodeIndex >= 0 && downNodeIndex < nodesCount();
```

Comment:
Hmm but I think that this assertion (and the above proof <3) is misplaced: even though you assert that before starting the `while` loops, this assertion theoretically might be no longer hold as the loops progress and decrease `downNodeIndex`/increase `upNodeIndex`...

Reply:
So I think the assertion either belongs to inside the `while` loops, or can be made a part of `while` loop conditions

Reply:
Possibly as a consequence you could get rid of warning suppression from this method